### PR TITLE
feat: use post for small files (client + sdk)

### PR DESCRIPTION
**Pull request by juliusmarminge**
needs https://github.com/pingdotgg/uploadthing-infra/pull/316

### CODE REVIEW

File: examples/minimal-appdir/src/server/uploadthing.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const uploadRouter = {
       acl: "public-read",
     },
     video: {
-      maxFileSize: "16MB",
```
Comment (markflorkowski) [inline examples/minimal-appdir/src/server/uploadthing.ts:29]: revert?

File: packages/uploadthing/src/internal/handler.ts [hunk 2]
```diff
@@ -120,18 +119,29 @@ export type RouterWithConfig<TRouter extends FileRouter> = {
...
-export type UploadThingResponse = {
-  presignedUrls: string[];
-  pollingJwt: string;
+type UploadThingBaseResponse = {
   key: string;
-  pollingUrl: string;
-  uploadId: string;
   fileName: string;
-  fileType: MimeType;
+  fileType: string;
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply packages/uploadthing/src/internal/handler.ts]: cause that wasn't the correct type (fileType here can be `image`, `blog` etc), I changed it to be `FileRouterInputKey` to cover those as well

File: examples/minimal-appdir/src/server/uploadthing.ts (hunk 1 above)
Comment (juliusmarminge) [reply examples/minimal-appdir/src/server/uploadthing.ts:29]: yea i just didn't have any file between 5 and 16 megs to test with haha

File: packages/uploadthing/src/client.ts [hunk 3]
```diff
@@ -312,3 +260,137 @@ export const generateClientDropzoneAccept = (fileTypes: string[]) => {
...
+  if (
+    presigned.fileType === file.type.split("/")[0] ||
+    presigned.fileType === file.type
+  ) {
+    formData.append("Content-Type", file.type);
+  } else if (presigned.fileType === "blob") {
+    formData.append("Content-Type", "application/octet-stream");
+  } else if (presigned.fileType === "pdf") {
+    formData.append("Content-Type", "application/pdf");
+  }
```
Comment (juliusmarminge) [inline packages/uploadthing/src/client.ts:345]: should this logic go on infra side and appended on the fields there?

Comment (markflorkowski) [reply packages/uploadthing/src/client.ts:345]: Yeah, not sure why we hadn't done it that way in the past

File: packages/uploadthing/src/constants.ts [hunk 4]
```diff
@@ -1 +1,4 @@
-export { version as UPLOADTHING_VERSION } from "../package.json";
+// export { version as UPLOADTHING_VERSION } from "../package.json";
+
+// FIXME: Revert before merge
+export const UPLOADTHING_VERSION = "6.4.0";
```
[the remark this answers is not in the document]
Comment (markflorkowski) [reply packages/uploadthing/src/constants.ts:4]: This is a smart way of doing this.

File: packages/uploadthing/src/client.ts [hunk 5]
```diff
@@ -312,3 +260,137 @@ export const generateClientDropzoneAccept = (fileTypes: string[]) => {
...
+    reportEventToUT: ReturnType<typeof createUTReporter>;
+    onUploadProgress?: UploadFilesOptions<any, any>["onUploadProgress"];
+  },
+) {
+  let etags: { tag: string; partNumber: number }[];
+  let uploadedBytes = 0;
+
+  try {
+    etags = await Promise.all(
+      presigned.presignedUrls.map(async (url, index) => {
```
Comment (markflorkowski) [inline packages/uploadthing/src/client.ts:277]: Any reason we don't destructure this higher?

Comment (juliusmarminge) [reply packages/uploadthing/src/client.ts:277]: no more than i didn't see a need and don't like destructuring unless i feel like i need to. if you think it's cleaner with it destructured happy to have it

**Review by markflorkowski — APPROVED** [review]

File: packages/uploadthing/src/client.ts (hunk 5 above)
Comment (juliusmarminge) [reply packages/uploadthing/src/client.ts:277]: Maybe rename field on infra to be just `url` and `urls`?

Comment (markflorkowski) [reply packages/uploadthing/src/client.ts:277]: `data` ?

Comment (markflorkowski) [reply packages/uploadthing/src/client.ts:277]: >Maybe rename field on infra to be just url and urls?

would be breaking, no?

Comment (juliusmarminge) [reply packages/uploadthing/src/client.ts:277]: Can do it just for this version?

Comment (juliusmarminge) [reply packages/uploadthing/src/client.ts]: e.g. like this https://github.com/pingdotgg/uploadthing-infra/pull/316/commits/57f5b2a0d5c715dd2eb028b23df78c993582bee0

**Pull request merged**